File: 2022/template.py
from pathlib import Path


def main() -> None:
    data = (Path(__file__).parent / "input.txt").read_text().strip()

    result_one = solve_one(data)
    print(result_one)
    assert result_one == ...

    result_two = solve_two(data)
    print(result_two)
    assert result_two == ...


def solve_one(data: str) -> ...:
    ...


def solve_two(data: str) -> ...:
    ...


if __name__ == "__main__":
    main()

File: 2025/template.go
package main

import (
	"fmt"
	"os"
	"strings"
)

const TODO int = -1

func main() {
	input := parseInput("input.txt")

	resultOne := solveOne(input)
	fmt.Println(resultOne)
	assert(resultOne == TODO, "solveOne()")

	resultTwo := solveTwo(input)
	fmt.Println(resultTwo)
	assert(resultTwo == TODO, "solveTwo()")
}

func solveOne(input string) int {
	return TODO
}

func solveTwo(input string) int {
	return TODO
}

func parseInput(filename string) string {
	content, err := os.ReadFile(filename)

	if err != nil {
		panic(fmt.Sprintf("Error reading file:", err))
	}

	return strings.TrimSuffix(string(content), "\n")
}

func assert(condition bool, message string) {
	if !condition {
		panic("Assertion failed: " + message)
	}
}
